### ISSUE TITLE
[BUG] HASHING_TEST fails with CUDA per-thread default stream

### ISSUE THREAD

**Issue by harrism** [rapidsai/cudf#2967]
**Describe the bug**

When libcudf is built with cmake flag `-DCMAKE_CUDA_FLAGS="--default-stream per-thread"`, 
HASHING_TEST fails.

**Steps/Code to reproduce bug**

From the build directory:

```
> cmake -DCMAKE_CUDA_FLAGS="--default-stream per-thread" ..
> gtests/HASHING_TEST
```

You'll get output like this:

```
[ RUN      ] HashPartitionTest/0.TenPartitions
../tests/hashing/hash_partition_test.cu:240: Failure
      Expected: partition_number
      Which is: 0
To be equal to: host_row_partition_numbers[i]
      Which is: 6
Partition number for row: 84680 doesn't match!
../tests/hashing/hash_partition_test.cu:240: Failure
      Expected: partition_number
      Which is: 0
To be equal to: host_row_partition_numbers[i]
      Which is: 5
Partition number for row: 84682 doesn't match!
../tests/hashing/hash_partition_test.cu:240: Failure
      Expected: partition_number
      Which is: 0
To be equal to: host_row_partition_numbers[i]
      Which is: 4
Partition number for row: 84684 doesn't match!
../tests/hashing/hash_partition_test.cu:240: Failure
      Expected: partition_number
      Which is: 0
To be equal to: host_row_partition_numbers[i]
      Which is: 5
Partition number for row: 84686 doesn't match!
```

**Expected behavior**

HASHING_TEST should pass

**Environment overview (please complete the following information)**

cuDF 0.10
DGX-1 (V100 32GB)

**Comment by jrhemstad** [comment]
The `gdf_hash_partition` test does make use of creating a stream in attempt to get some overlap between two kernels. I quickly experimented to see if removing this extra stream fixed the issue and it did not. It was not immediately obvious to me why this would fail.

**Comment by harrism** [comment]
With the new libcudf++ implementation HASHING_TEST no longer fails with per-thread default stream, while LEGACY_HASHING_TEST (old implementation) still fails. Closing this as fixed by #3144
